### DISCUSSION TITLE
How to keep DBAPI connections limited when using async_scoped_session

### DISCUSSION THREAD

**Discussion by pankajkumar229** [sqlalchemy/sqlalchemy#9097] [Usage Questions] [answered]
We are starting to use async_scoped_session with the session scope of current_task. 

The application receives requests that is served by a tornado async handler, it queries/changes database and then calls close() and remove() as a finalizer. 

The number of DBAPI connections seems to keep increasing during the entire lifetime of the application even if pool_size is specified. It seems to hang after using too many connections. When all transactions are done, the number of DBAPI connections is still high. 

I am not sure engine.dispose can be called because it may trouble in flight transactions.

What is the correct way of keeping DBAPI connections to a limit?

**Comment by CaselIT** [comment, marked as answer]
Hi,

That's not the sqlalchemy normal behaviour. Ensure that close and remove is called at the end of each request

**Comment by pankajkumar229** [comment]
Thanks, There were a few exceptions not closing the session properly
